### ISSUE TITLE
Vulkan: Mipmap textures not rendering correctly on integrated GPU

### ISSUE THREAD

**Issue by deadpixelsociety** [godotengine/godot#85348]
### Godot version

4.2.rc2

### System information

Windows 11, Vulkan (Forward+), Intel Iris Xe Graphics

### Issue description

When generating mipmaps for textures and setting the Linear Mipmap texture filter default some textures are not rendered correctly on systems with integrated GPUs.
 
This has affected multiple users when playing our game on a system with an integrated GPU on various devices. I'm able to replicate on my laptop here using Intel Iris Xe Graphics with the MRP attached.

**Desktop (Geforce RTX 2060)**
![image](https://github.com/godotengine/godot/assets/6668682/1d33436a-27ad-40ce-8dc0-9f313d478217)

**Laptop (Intel Iris Xe Graphics)**
![image](https://github.com/godotengine/godot/assets/6668682/e80073fd-0de8-494f-8397-e490ffbfe730)

### Steps to reproduce

1. Add textures to project, generate mipmaps for those textures.
2. Enable Linear Mipmap texture filter default in project settings.
3. Add textures to scene as sprites.

### Minimal reproduction project

[VulkanTest.zip](https://github.com/godotengine/godot/files/13464433/VulkanTest.zip)

**Comment by deadpixelsociety** [comment]
A peer noticed the issue and I've confirmed it on my side now as well. The lemon texture used above is an odd size, while the apple texture is even. So, it looks like odd-sized textures using mipmap filtering on integrated GPUs produces this issue.

**Comment by Chaosus** [comment]
Yes, old GPU's being known for supporting only the textures with a size with a power of 2. I guess, there is nothing we can do about it other than maybe provide a warning when the Godot editor found that any texture violate this rule.

**Comment by Calinou** [comment]
> Yes, old GPU's being known for fully supporting only the textures with a size of power of 2. I guess, there is nothing we can do about it other than maybe provide a warning when the Godot editor found that any texture violate this rule.

NPOT texture support is guaranteed since OpenGL ES 3.0. Iris Xe is much newer than that, so it's not related.
